app/src/org/commcare/activities/connect/HQUserInviteActivity.java

```diff
@@ -0,0 +1,173 @@
+            binding.tvHqInvitationHeaderTitle.setText(isTokenPresent
+                    ? getString(R.string.connect_hq_invitation_heading, username)
+                    : getString(R.string.connect_hq_invitation_connectId_not_configure));
+            setButtonVisibility(isTokenPresent);
+            setButtonListeners(isTokenPresent);
+        } else {
+            binding.tvHqInvitationHeaderTitle.setText(getString(R.string.connect_hq_invitation_wrong_user));
+            setButtonVisibility(false);
+        }
+    }
+
+    private void setButtonVisibility(boolean visible) {
+        binding.btnAcceptInvitation.setVisibility(visible ? View.VISIBLE : View.GONE);
+        binding.btnDeniedInvitation.setVisibility(visible ? View.VISIBLE : View.GONE);
+        binding.btnGoToRecovery.setVisibility(visible ? View.GONE : View.VISIBLE);
+    }
+
+    private void setButtonListeners(boolean isTokenPresent) {
+        if (isTokenPresent) {
+            binding.btnAcceptInvitation.setOnClickListener(view -> handleInvitation(callBackURL, inviteCode));
+            binding.btnDeniedInvitation.setOnClickListener(view -> declineInvitation());
+        } else {
+            binding.btnGoToRecovery.setOnClickListener(view -> ConnectManager.registerUser(this, success -> {
+                        if (success) {
+                            ConnectManager.goToConnectJobsList(this);
+                        }
+                    })
+            );
+        }
+    }
+
+    private void declineInvitation() {
+        FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, false, "");
+        finish();
+    }
+
+    private void handleInvitation(String callBackUrl, String inviteCode) {
+        Logger.log("HQInvite", "User accepted invitation");
+
+        IApiCallback callback = new IApiCallback() {
+            @Override
+            public void processSuccess(int responseCode, InputStream responseData) {
+                FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, true, "");
+                Logger.log("HQInvite", "Acceptance succeeded");
+                binding.progressBar.setVisibility(View.GONE);
+                try {
+                    String responseAsString = new String(StreamsUtil.inputStreamToByteArray(responseData));
+                    if (responseAsString.length() > 0) {
+                        startActivity(new Intent(HQUserInviteActivity.this, DispatchActivity.class));
+                        finish();
+                    }
+                } catch (IOException e) {
+                    CrashUtil.reportException(e);
+                    setErrorMessage(getString(R.string.connect_hq_invitation_accept_error));
+                }
+            }
+
+            @Override
+            public void processFailure(int responseCode, IOException e) {
+                FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, false, "API error");
+                Logger.log("HQInvite", "Acceptance failed");
```

Comment:
probably log the IOException as Logger.exception() instead if it's not happening in the network layer itself.